File: rust/src/models/list_response_integration_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::integration_out::IntegrationOut;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ListResponseIntegrationOut {
    pub data: Vec<IntegrationOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(rename = "prevIterator")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseIntegrationOut {
    pub fn new(data: Vec<IntegrationOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }
}

File: javascript/src/models/listResponseIntegrationOut.ts
// this file is @generated
import { type IntegrationOut, IntegrationOutSerializer } from "./integrationOut";

export interface ListResponseIntegrationOut {
  data: IntegrationOut[];
  iterator: string | null;
  prevIterator?: string | null;
  done: boolean;
}

export const ListResponseIntegrationOutSerializer = {
  _fromJsonObject(object: any): ListResponseIntegrationOut {
    return {
      data: object["data"].map((item: IntegrationOut) =>
        IntegrationOutSerializer._fromJsonObject(item)
      ),
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
      done: object["done"],
    };
  },

  _toJsonObject(self: ListResponseIntegrationOut): any {
    return {
      data: self.data.map((item) => IntegrationOutSerializer._toJsonObject(item)),
      iterator: self.iterator,
      prevIterator: self.prevIterator,
      done: self.done,
    };
  },
};
